frontend/src/metabase/dashboard/actions/tabs.ts

```diff
@@ -416,27 +433,34 @@ export const tabsReducer = createReducer<DashboardState>(
       },
     );
 
-    builder.addCase(undoDeleteTab, (state, { payload: { tabDeletionId } }) => {
-      const { prevTabs } = getPrevDashAndTabs({ state });
-      const { tabId, removedDashCardIds } = state.tabDeletions[tabDeletionId];
-      const removedTab = prevTabs.find(({ id }) => id === tabId);
-      if (!removedTab) {
-        throw new Error(
-          `UNDO_DELETE_TAB was dispatched but tab with id ${tabId} was not found`,
-        );
-      }
+    builder.addCase(
+      undoDeleteTab,
+      (state, { type, payload: { tabDeletionId } }) => {
+        const { prevTabs } = getPrevDashAndTabs({ state });
+        const { tabId, removedDashCardIds } = state.tabDeletions[tabDeletionId];
+        const removedTab = prevTabs.find(({ id }) => id === tabId);
+        if (!removedTab) {
+          throw new Error(
+            `UNDO_DELETE_TAB was dispatched but tab with id ${tabId} was not found`,
+          );
+        }
 
-      // 1. Unmark tab as removed
-      removedTab.isRemoved = false;
+        markDashboardDirty(type, state);
```

Comment:
this line is the only change in this fn, everything else is due to indentation change from destructuring `type` from the fn's args